server/api/middleware.go

```diff
@@ -86,3 +88,57 @@ type clusterCtxKey struct{}
 func getCluster(r *http.Request) *cluster.RaftCluster {
 	return r.Context().Value(clusterCtxKey{}).(*cluster.RaftCluster)
 }
+
+type auditMiddleware struct {
+	svr *server.Server
+}
+
+func newAuditMiddleware(s *server.Server) negroni.Handler {
+	return &auditMiddleware{svr: s}
+}
+
+// ServeHTTP is used to implememt negroni.Handler for auditMiddleware
+func (s *auditMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if !s.svr.IsServiceMiddlewareEnabled() || !s.svr.IsAuditMiddlewareEnabled() {
+		next(w, r)
+		return
+	}
+
+	requestInfo, ok := requestutil.RequestInfoFrom(r.Context())
+	if !ok {
+		next(w, r)
+	}
+
+	labels := s.svr.GetServiceAuditBackendLabels(requestInfo.ServiceLabel)
+	if labels == nil {
+		next(w, r)
+		return
+	}
+
+	failpoint.Inject("addAuditMiddleware", func() {
+		w.Header().Add("audit-label", strings.Join(labels.Labels, ","))
+	})
+
+	beforeNextBackends := make([]audit.Backend, 0)
+	afterNextBackends := make([]audit.Backend, 0)
+	for _, backend := range s.svr.GetAuditBackend() {
+		if backend.Match(labels) {
+			if backend.ProcessBeforeHandler() {
+				beforeNextBackends = append(beforeNextBackends, backend)
+			} else {
+				afterNextBackends = append(afterNextBackends, backend)
+			}
+		}
+	}
+	for _, backend := range beforeNextBackends {
+		backend.ProcessHTTPRequest(r)
+	}
```

Comment:
```suggestion
for _, backend := range s.svr.GetAuditBackend() {
    if backend.Match(labels) && backend.ProcessBeforeHandler() {
                  backend.ProcessHTTPRequest(r)
    }
}
```

Reply:
The backend can be divided into two types: before the service Handle and after the service Handle